bitcoin/contrib/test.sh

```diff
@@ -94,16 +99,24 @@ then
     )
 fi
 
+# Run formatter if told to.
+if [ "$DO_FMT" = true ]; then
+    if [ "$NIGHTLY" = false ]; then
+        echo "DO_FMT requires a nightly toolchain (consider using RUSTUP_TOOLCHAIN)"
+        exit 1
+    fi
+    rustup component add rustfmt
+    cargo fmt --check
```

Comment:
This is wrong, should be `cargo fmt --check || exit 1`.

Reply:
Because we have `set -e` at the top I don't think this is necessary. Though I wouldn't mind it either, since it makes things clearer.

Reply:
oh yes, of course. My bad.

Reply:
I left it as is because non of the other commands explicitly exit, changing this would require me to change all instances across all our CI scripts else I might start twitching when I next read them.